numpy/core/tests/test_multiarray.py

```diff
@@ -864,8 +864,12 @@ def test_casting(self):
 
         # Check that 'equiv' casting can reorder fields and change byte
         # order
+        # New in 1.12: This behavior changes in 1.13, test for dep warning
         assert_(np.can_cast(a.dtype, b.dtype, casting='equiv'))
-        c = a.astype(b.dtype, casting='equiv')
+        with warnings.catch_warnings(record=True) as log:
```

Comment:
actually, maybe assert_warns even was a context manager by now, I somewhat think it was....